File: GUIA/d29_main_application_project/monitor_omega/monitor_omega-db/src/main/java/mx/com/telcel/di/sds/gsac/dsc/ws/main_application/monitor_omega/db/model/ReportePantallaOmega.java
package mx.com.telcel.di.sds.gsac.dsc.ws.main_application.monitor_omega.db.model;

import java.util.Date;

/**
 * Clase con la estructura de la pantalla del monitor Omegamon II for DB2
 *
 * @author VI9XXI3
 */
public class ReportePantallaOmega {

    private Date lastUpdate;
    private int alertsCount;
    private OmegaData omegaData;
    private PropiedadesMonitorOmega propiedadesMonitorOmega;

    public ReportePantallaOmega() {
    }

    public Date getLastUpdate() {
        return lastUpdate;
    }

    public void setLastUpdate(Date lastUpdate) {
        this.lastUpdate = lastUpdate;
    }

    public int getAlertsCount() {
        return alertsCount;
    }

    public void setAlertsCount(int alertsCount) {
        this.alertsCount = alertsCount;
    }

    public OmegaData getOmegaData() {
        return omegaData;
    }

    public void setOmegaData(OmegaData omegaData) {
        this.omegaData = omegaData;
    }

    public PropiedadesMonitorOmega getPropiedadesMonitorOmega() {
        return propiedadesMonitorOmega;
    }

    public void setPropiedadesMonitorOmega(PropiedadesMonitorOmega propiedadesMonitorOmega) {
        this.propiedadesMonitorOmega = propiedadesMonitorOmega;
    }
}

File: GUIA/d29-main-application/src/app/monitor-omega/model/ReportePantallaOmega.ts
import {OmegaData, toOmegaData} from './OmegaData';
import {PropiedadesMonitorOmega, toPropiedadesMonitorOmega} from './PropiedadesMonitorOmega';

export class ReportePantallaOmega {
  constructor(
    public lastUpdate: Date,
    public alertsCount: number,
    public omegaData: OmegaData,
    public propiedadesMonitorOmega: PropiedadesMonitorOmega
  ) {}

  static getNewInstance(): ReportePantallaOmega {
    return new ReportePantallaOmega(new Date(), 0, OmegaData.getNewInstance(), PropiedadesMonitorOmega.getNewInstance());
  }
}

export function mapPantalla(response: any): ReportePantallaOmega {
  let pantalla: ReportePantallaOmega = ReportePantallaOmega.getNewInstance();
  let responseJson = response.json();
  if (responseJson) {
    pantalla = <ReportePantallaOmega>({
      lastUpdate: responseJson.lastUpdate,
      alertsCount: responseJson.alertsCount,
      omegaData: toOmegaData(responseJson.omegaData),
      propiedadesMonitorOmega: toPropiedadesMonitorOmega(responseJson.propiedadesMonitorOmega)
    });
  }
  return pantalla;
}
